mui/mui.py

```diff
@@ -136,27 +203,155 @@ def solve(bv: BinaryView):
         )
         return
 
-    # Start a solver thread for the path associated with the view
-    s = ManticoreRunner(bv.session_data.mui_find, bv.session_data.mui_avoid, bv)
-    s.start()
+    dialog = RunDialog(DockHandler.getActiveDockHandler().parent(), bv)
+    result: QDialog.DialogCode = dialog.exec()
+
+    if result == QDialog.Accepted:
+        # Start a solver thread for the path associated with the view
+        bv.session_data.mui_is_running = True
+        s = ManticoreRunner(bv.session_data.mui_find, bv.session_data.mui_avoid, bv)
+        s.start()
+
+
+def stop_manticore(bv: BinaryView):
+    """Stops the current running manticore instance"""
+    bv.session_data.mui_is_running = False
+
+
+def avoid_instr_is_valid(bv: BinaryView, addr: int):
+    """checks if avoid_instr is valid for a given address"""
+    return addr not in bv.session_data.mui_avoid
+
+
+def find_instr_is_valid(bv: BinaryView, addr: int):
+    """checks if find_instr is valid for a given address"""
+    return addr not in bv.session_data.mui_find
+
+
+def solve_is_valid(bv: BinaryView):
+    """checks if solve is valid for a given binary view"""
+    return not bv.session_data.mui_is_running
 
 
 PluginCommand.register_for_address(
     "MUI \\ Find Path to This Instruction",
     "When solving, find a path that gets to this instruction",
     find_instr,
+    find_instr_is_valid,
+)
+PluginCommand.register_for_address(
+    "MUI \\ Remove Instruction from Find List",
+    "When solving, DO NOT find paths that reach this instruction",
+    rm_find_instr,
+    lambda bv, addr: not find_instr_is_valid(bv, addr),
 )
 PluginCommand.register_for_address(
     "MUI \\ Avoid This Instruction",
     "When solving, avoid paths that reach this instruction",
     avoid_instr,
+    avoid_instr_is_valid,
+)
+PluginCommand.register_for_address(
+    "MUI \\ Remove Instruction from Avoid List",
+    "When solving, DO NOT avoid paths that reach this instruction",
+    rm_avoid_instr,
+    lambda bv, addr: not avoid_instr_is_valid(bv, addr),
 )
 PluginCommand.register(
     "MUI \\ Solve With Manticore",
     "Attempt to solve for a path that satisfies the constraints given",
     solve,
+    solve_is_valid,
+)
+PluginCommand.register(
+    "MUI \\ Stop Manticore",
+    "Stop the running manticore instance",
+    stop_manticore,
+    lambda bv: not solve_is_valid(bv),
 )
 
 widget.register_dockwidget(
     StateListWidget, StateListWidget.NAME, Qt.RightDockWidgetArea, Qt.Vertical, True
 )
+
+widget.register_dockwidget(
+    StateGraphWidget, StateGraphWidget.NAME, Qt.TopDockWidgetArea, Qt.Vertical, True
+)
+
+settings = Settings()
+if not settings.contains("mui.run_argv"):
+    settings.register_group("mui", "MUI Settings")
```

Comment:
Ahhh so we'll need a few module constants. It might be best to build up the module constant for the settings prefix like the following (use whatever naming you like):
```python
BINJA_SETTINGS_GROUP: Final[str] = "mui"
BINJA_SETTINGS_PREFIX: Final[str] = f"{BINJA_SETTINGS_GROUP}.run."
```